ksql-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java

```diff
@@ -141,23 +141,15 @@ public KeyField getKeyField() {
       final SchemaKStream schemaKStream,
       final QueryContext.Stacker contextStacker
   ) {
-
     if (schemaKStream instanceof SchemaKTable) {
       return schemaKStream;
     }
 
-    final KeyField resultKeyField = KeyField.of(
-        schemaKStream.getKeyField().ref(),
-        getKeyField().legacy()
-    );
-
-    final SchemaKStream result = schemaKStream.withKeyField(resultKeyField);
```

Comment:
`withKeyField` has been removed as its no longer needed.  The only reason it was needed was to set the legacy (incorrect) keyField.  With this PR this all goes away.